csrc/device_lower/pass/expr_sort.cpp

```diff
@@ -1114,26 +1115,26 @@ void ExprSegmentationSorter::initializeForLoopDependencies() {
     visited.emplace(id);
   }
   if (failed) {
-    std::cerr
```

Comment:
Most changes were mechanical, but note that in a few places `std::cerr` was being used instead of `std::cout`. If we move away from using `TORCH_CHECK` and friends, we should consider creating another stream guard like `ErrorStreamGuard` and accessor `nvferror()`, so that we can capture error information to present in Python exceptions. For now, we may want to even introduce the nvferror() method and have it just return std::cerr, or leave these replacements alone..

Reply:
Most uses of `std::cerr` are for regular debugging output, so I changed those to use `nvfdebug()`. There is one case (in `ExprSegmentationSorter`) where `std::cerr` was written to before raising an exception with no message. Instead, I use a `stringstream` to build the exception message. This way the exception message should be captured in Python as well. I don't think there is any more explicit `std::cerr` output in nvfuser at the moment, so we can probably avoid having a separate err stream at this time.